scanners/reactnative-expo/reactnative-expo.go

```diff
@@ -0,0 +1,706 @@
+
+	fmt.Printf("Returning true for expo")
+	iosScanner := ios.NewScanner()
+	androidScanner := android.NewScanner()
+	for _, packageJSONPth := range packageJSONPths {
+		log.TPrintf("checking: %s", packageJSONPth)
+
+		projectDir := filepath.Dir(packageJSONPth)
+
+		iosDir := filepath.Join(projectDir, "ios")
+		if exist, err := pathutil.IsDirExists(iosDir); err != nil {
+			return false, err
+		} else if exist {
+			if detected, err := iosScanner.DetectPlatform(scanner.searchDir); err != nil {
+				return false, err
+			} else if detected {
+				return false, nil
+			}
+		}
+
+		androidDir := filepath.Join(projectDir, "android")
+		if exist, err := pathutil.IsDirExists(androidDir); err != nil {
+			return false, err
+		} else if exist {
+			if detected, err := androidScanner.DetectPlatform(scanner.searchDir); err != nil {
+				return false, err
+			} else if detected {
+				return false, nil
+			}
+		}
+	}
+
+	if err := ensureNodeModules(); err != nil {
+		log.Errorf("ERROR DURING INSTALLING NPM: %s", err)
+		return false, nil
+	}
+
+	if err := ejectProject(searchDir); err != nil {
+		log.Errorf("ERROR DURING EJECTING THE PROJECT: %s", err)
+		return false, nil
+	}
+
+	return scanner.reactnativeScannerDetectPlatform(scanner.searchDir)
+}
+
+func (scanner *Scanner) reactnativeScannerDetectPlatform(searchDir string) (bool, error) {
+	scanner.searchDir = searchDir
+
+	log.TInfof("Collect package.json files")
+
+	packageJSONPths, err := reactnative.CollectPackageJSONFiles(searchDir)
+	if err != nil {
+		return false, err
+	}
+
+	log.TPrintf("%d package.json file detected", len(packageJSONPths))
+
+	log.TInfof("Filter relevant package.json files")
+
+	relevantPackageJSONPths := []string{}
+	iosScanner := ios.NewScanner()
+	androidScanner := android.NewScanner()
+	for _, packageJSONPth := range packageJSONPths {
+		log.TPrintf("checking: %s", packageJSONPth)
+
+		projectDir := filepath.Dir(packageJSONPth)
+
+		iosProjectDetected := false
+		iosDir := filepath.Join(projectDir, "ios")
+		if exist, err := pathutil.IsDirExists(iosDir); err != nil {
+			return false, err
+		} else if exist {
+			if detected, err := iosScanner.DetectPlatform(scanner.searchDir); err != nil {
+				return false, err
+			} else if detected {
+				iosProjectDetected = true
+			}
+		}
+
+		androidProjectDetected := false
+		androidDir := filepath.Join(projectDir, "android")
+		if exist, err := pathutil.IsDirExists(androidDir); err != nil {
+			return false, err
+		} else if exist {
+			if detected, err := androidScanner.DetectPlatform(scanner.searchDir); err != nil {
+				return false, err
+			} else if detected {
+				androidProjectDetected = true
+			}
+		}
+
+		if iosProjectDetected || androidProjectDetected {
+			relevantPackageJSONPths = append(relevantPackageJSONPths, packageJSONPth)
+		} else {
+			log.TWarnf("no ios nor android project found, skipping package.json file")
+		}
+	}
+
+	if len(relevantPackageJSONPths) == 0 {
+		return false, nil
+	}
```

Comment:
if more then 1 package.json file found list them and log using warning level which one will we use